docs/sources/mimir/visualize/query-metric-labels.md

```diff
@@ -0,0 +1,95 @@
+  `GET,POST <prometheus-http-prefix>/api/v1/cardinality/label_names`
+
+> **Note**: `<prometheus-http-prefix>` is the Prometheus HTTP prefix as documented in the [HTTP API]({{< relref "../references/http-api/index.md#path-prefixes" >}}) reference.
+
+### Features comparison
+
+The different API endpoints have different features.
+The following list summarises the main differences.
+
+#### Can specify the time range in the request?
+
+- **Get label names**<br />
+  Yes. However, the actual time range granularity is restricted to TSDB block range periods (2h for the most recent metrics and up to 24h for historical metrics when running Mimir with the default configuration). In practice, the input `start` and `end` parameters are rounded to the TSDB block boundaries: the `start` parameter is rounded to the start of the nearest block containing samples whose timestamp is lower or equal than the input timestamp, while the `end` parameter is rounded to the end of the nearest block containing samples with timestamp greater or equal than the input timestamp.
+- **Label names cardinality**<br />
+  No. The API queries only the in-memory series in the ingester.
+
+#### Can specify the series selector in the request?
+
+- **Get label names API**<br />
+  Yes. The optional `match[]` parameter allows to select the series from which to read the label names. The `match[]` parameter can be specified multiple times. When omitted, label names are extracted from all series.
+- **Label names cardinality API**<br />
+  Yes. The optional `selector` parameter allows to select the series from which to read the label names. The `selector` parameter can be specified only once. When omitted, label values are extracted from all series.
+
+### Recommendations and performance trade-off
+
+- Prefer "get label names" over "label names cardinality" API, unless you need to know the number of unique label values for each label name.
+- The "get label names" API `start` and `end` parameters are optional. When omitted, Mimir will query label names for the whole retention period. We recommend to always specify `start` and `end` parameters, and preferably set to the shortest period which is feasible for your use case.
+- For both API endpoints, executing a request specifying the series selector is computationally more expensive than the same request without the series selector, but the results set can be significantly smaller when the series selector effectively reduce the series set.
+
+## Querying label values
+
+The following API endpoints can be used to find the list of label values for a given label name:
+
+- **[Get label values]({{< relref "../references/http-api/index.md#get-label-values" >}})**<br />
+  `GET <prometheus-http-prefix>/api/v1/label/{name}/values`
+- **[Label values cardinality]({{< relref "../references/http-api/index.md#label-values-cardinality" >}})**<br />
+  `GET,POST <prometheus-http-prefix>/api/v1/cardinality/label_values`
```

Comment:
Thanks very much @pracucci for putting this together - It seems like a nice addition to our docs.

To me this doesn't feel like a complete list yet. Why is there no mention of:
- `/series` (or with a matcher, `series?match[]=grafanacloud_org_info{org_slug="grafana"}`? This seems to be how Prom Datasource Dashboard variables of **Type: Query, Query-type: Label values** seems to be implemented in Grafana 10.0. I think it's worth explicitly mentioning this option, even if you want to simply say "Don't use this, we're deprecating it". (It could be nice to say **why** it's being deprecated somewhere)
- `/query`? Which is what I had to switch to recently when `/label/{name}/values` was too slow. I ended up using `sum by(org_id) (grafanacloud_org_info{org_slug="$org_slug"})` with a regex `/org_id="(?<value>[^"]+)/g`

It seems to be both of these options are also viable candidates to help our users to query for the data that they ultimately want and need, and should be considered as alternatives here.

Reply:
> Grafana is switching from /series to "label values"? That's how I understood issue that you've reported internally.

Yes. Switching from `/series` in 10.0 to "label values" in 10.1

Reply:
I've improved the doc based on your suggestions, while doing my best to keep it lightweight.